src/test/java/com/google/devtools/build/lib/remote/RemoteSpawnCacheTest.java

```diff
@@ -503,4 +503,29 @@ public Void answer(InvocationOnMock invocation) {
         .containsExactly(Pair.of(ProgressStatus.CHECKING_CACHE, "remote-cache"));
     assertThat(eventHandler.getEvents()).isEmpty(); // no warning is printed.
   }
+
+  @Test
+  public void failedCacheActionAsCacheMiss() throws Exception {
+    ActionResult actionResult = ActionResult.newBuilder().setExitCode(1).build();
+    when(remoteCache.getCachedActionResult(any(ActionKey.class)))
+        .thenAnswer(
+            (Answer<ActionResult>) invocation -> {
+              RequestMetadata meta = TracingMetadataUtils.fromCurrentContext();
+              assertThat(meta.getCorrelatedInvocationsId()).isEqualTo("build-req-id");
+              assertThat(meta.getToolInvocationId()).isEqualTo("command-id");
+              return actionResult;
+            });
+
+    CacheHandle entry = cache.lookup(simpleSpawn, simplePolicy);
+
+    assertThat(entry.hasResult()).isFalse();
```

Comment:
why do you think the asserts below this line are valuable for the test?

Reply:
We need to verify, that current action indeed treated as a cash miss => we do not attempt to download anything. 
Asserts in line 528-529 are obsolete, removing them.

Reply:
Whether or not something is being downloaded seems orthogonal to what should be tested here. I'd argue that all this test should verify is that it's indeed a cache miss. We then have a separate unit test that verifies what should/should not happen on a cache miss.

Reply:
Fair point. Removed.